### PR TITLE
save1

### PR DESCRIPTION
**Pull request by MikalaiHud**

### CODE REVIEW

File: src/java/main/com/epam/hud/Main.java [hunk 1]
```diff
@@ -1,181 +1,71 @@
...
+import com.epam.hud.dao.interfaces.ShopDAO;
+import com.epam.hud.dao.interfaces.ToyDAO;
+import com.epam.hud.generators.ShopGenerators;
+import com.epam.hud.generators.ToyGenerators;
+import com.epam.hud.logic.ShopManager;
+
+import java.sql.SQLException;
+
+
+// TODO: Проверить форматирование кода в каждом файле, убрать не используемые импорты. Добавить пример использования методов в Main.
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/Main.java]: Why you didn't remove comments?

File: src/java/main/com/epam/hud/Main.java [hunk 2]
```diff
@@ -1,181 +1,71 @@
...
-        //класса AnimeToy.
-        //Создание объекта игрушки
-
-        AnimeToy animeToy = new AnimeToy();
-        animeToy.setToyFandom(Fandom.NARUTO);
-        animeToy.setToyName("Uzumaki");
-        animeToy.setToyType(Type.ACTION_FIGURE);
-        animeToy.setToyPrice(20);
-        System.out.println(animeToy);
+    public static void main(String[] args) throws SQLException {
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/Main.java]: throws SQLException - ? 
add try-catch block for handle exceptions

File: src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        statement.setString(2, animeShop.getShopAddress());
+        statement.executeUpdate();
+        connection.close();
+    }
+
+
+    public AnimeShop readShop(int id) throws SQLException {
+        AnimeShop shop = new AnimeShop(); // Making object of class AnomeShop to set necessary data there.
+        Connection connection = ConnectionPool.getInstance().getConnection();
+        PreparedStatement statement = connection.prepareStatement(READ_SHOP); //
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java]: add try-catch block and use finally block for closing connection
also need to add preparedStatement close()

File: src/java/main/com/epam/hud/file/FileWorker.java [hunk 4]
```diff
@@ -19,9 +19,10 @@ public void writeShops(String filename, ShopContainer shopContainer) throws IOEx
     public ArrayList<AnimeShop> readShops(String filename) throws IOException, ClassNotFoundException {
         FileInputStream fis = new FileInputStream(filename);
         ObjectInputStream ois = new ObjectInputStream(fis);
-        ShopContainer shopContainer = (ShopContainer) ois.readObject();
+        ShopManager shopContainer = (ShopManager) ois.readObject();
         ois.close();
         fis.close();
-        return shopContainer.getShops();
+        //shopContainer.getShops();
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/file/FileWorker.java]: remove unnecessary comments

File: src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java [hunk 5]
```diff
@@ -16,68 +16,146 @@
     private final String DELETE_SHOP = "DELETE from anime.shops where id=?";
     private final String READ_ALL_SHOPS = "SELECT * from anime.shops";
 
+    // TODO: Прочитать про try/ try-with resources.
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java]: unnecessary comment

File: src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java [hunk 6]
```diff
@@ -16,68 +16,146 @@
...
-        statement.setString(2, animeShop.getShopAddress());
-        statement.setInt(3, id);
-        statement.executeUpdate();
-
-    }
+        @Override
+        public void updateShop ( int id, AnimeShop animeShop)  {
+            Connection connection = ConnectionPool.getInstance().getConnection();
+            PreparedStatement statement = null;
+            try {
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/dao/implement/ShopDAOImpl.java]: Bad solution for DAO impl. Make better with your own exception classes

File: src/java/main/com/epam/hud/logic/ShopManager.java [hunk 7]
```diff
@@ -24,50 +24,50 @@
...
 
-    public String getShopNameById(int id) throws SQLException {
+    public String getShopNameById(int id) {
         ShopDAO shopDAO = new ShopDAOImpl();
         AnimeShop shop = shopDAO.readShop(id);
         return shop.getShopName();
     }
 
-    public ArrayList<AnimeToy> getAllToysFromDB() throws SQLException {
+    public ArrayList<AnimeToy> getAllToysFromDB() {
```
Comment (artsiom-bukhautsou) [inline src/java/main/com/epam/hud/logic/ShopManager.java]: Make simple names for methods. For example: getAllToys, getCheapestToy

**Review by artsiom-bukhautsou — APPROVED** [review]

**Comment by bukhavtsov** [comment]
Final task has been completed
Well done Mikalai!

**Pull request merged**